Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -393,16 +393,33 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner, Cancellable {
 
         // Optionally wrap the command in a sandbox, which places some limits on what it can do. In particular, it blocks network access and restricts the paths to which the plugin can make file system changes. It does allow writing to temporary directories.
         if self.enableSandbox {
-            command = Sandbox.apply(command: command, strictness: .writableTemporaryDirectory, writableDirectories: writableDirectories + [self.cacheDir], readOnlyDirectories: readOnlyDirectories)
+            let sandboxProfile = SandboxProfile(
+                // Allow writing in the temporary directory (whether or not it's overridden by TMPDIR).
+                [localFileSystem.tempDirectory].map{ .writable($0) } +
+
+                // But prevent writing in any read-only directories.
+                readOnlyDirectories.map{ .readonly($0) } +
+
+                // But allow writing in any writable directories.
+                writableDirectories.map{ .writable($0) } +
+
+                // And always allow writing to the cache directory, even if it is inside one of the writable directories.
```

Comment:
writable => read-only, right?

Reply:
It's in order actually, so last one wins.